### ISSUE TITLE
Disappointing Converter performance

### ISSUE THREAD

**Issue by hinerm** [scijava/scijava#197]
I am starting with this 16-bit image:
<img width="685" alt="image" src="https://github.com/scijava/scijava/assets/1123809/129555b1-8250-4a73-b17e-1ffa587e38e3">

If I convert it with `Image > Type > 8-bit` in Fiji I get:
<img width="682" alt="image" src="https://github.com/scijava/scijava/assets/1123809/236dfedd-b6a4-4a55-ae9a-e5790d44834d">

If I denoise this image I get this result (which I'm happy with):
<img width="679" alt="image" src="https://github.com/scijava/scijava/assets/1123809/0adc3d19-7c01-4ca7-b210-399c32dbf98f">

But if I try to convert it with e.g. `result = ops.unary("convert.int8").input(img).apply()` I get.. a `32-bit` image??
<img width="687" alt="image" src="https://github.com/scijava/scijava/assets/1123809/eb10677c-efb4-4c47-819e-70dc0ef0843f">

`convert.uint8` is 8-bit, at least..
<img width="681" alt="image" src="https://github.com/scijava/scijava/assets/1123809/8597583e-2962-451c-b7d3-b2b4abd5e788">

Smoothing the uint-8 version requires bumping up the smoothing strength significantly and it still doesn't look great:
<img width="691" alt="image" src="https://github.com/scijava/scijava/assets/1123809/3e13ae29-e027-4dbb-8b22-6457236e080b">

So, how I account for this behavior is:
- In the ImageJ type conversion, values are scaled proportionately to the new range
- In the SJ Ops type conversion, values are directly translated. This can result in type overflow (note the split tails in both the `int8` and `uint8` conversions) which **fundamentally changes the behavior of subsequent image processing**.
- These [byte casts](https://github.com/scijava/incubator/blob/a969114d2f047a47dffb3ca2d189a3263d56d98a/scijava-ops-image/src/main/java/org/scijava/ops/image/convert/ConvertTypes.java#L111-L125) are perhaps not working as intended?

@gselzer @ctrueden Is there rationale for not scaling the converters to the target data range?

**Comment by gselzer** [comment]
> That should be a different op, in my opinion—it's not just a type conversion anymore at that point. IIRC, ImageJ Ops has a `convert.scale` but the design of it wasn't super great, so we should probably rediscuss how we want these sorts of operations to work.

Yeah, I do agree that the best way forward is to make separate Ops for scale vs cast vs clamp, because this is how it worked in ImageJ Ops. I'd also note that NumPy's main mechanism for converting data types, [numpy.ndarray.astype](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html), also casts without scaling:
```python
>>> np.array([10000000], dtype=np.int32).astype(np.int8)
array([-128], dtype=int8)
>>> np.array([10000001], dtype=np.int32).astype(np.int8)
array([-127], dtype=int8)
>>> np.array([10000002], dtype=np.int32).astype(np.int8)
array([-126], dtype=int8)
```

**Comment by hinerm** [comment]
I was looking around and OpenCV does have a set of [normalize methods](https://github.com/bytedeco/javacpp-presets/blob/351d203dece22db179ffcfb63ebeee548ef18a14/opencv/src/gen/java/org/bytedeco/opencv/global/opencv_core.java#L8318-L8326) that we could wrap to Ops.

Caveats:
- They have optional parameters which I'm pretty sure wouldn't be captured
- Conversion to `Mats` is limited and requires lots of copying.

**Comment by hinerm** [comment]
Current plan:
- [ ] Add normalize ops, e.g. see [those available in OpenCV](https://docs.opencv.org/4.x/d2/de8/group__core__array.html#gad12cefbcb5291cf958a85b4b67b6149f). Written as a computer that takes an `I` input and `O` output and uses the `O` type min/max for normalizing. 
- [ ] Rename current type convert ops to be called `cast`
- [ ] Possibly refactor [current `convert.clip`](https://github.com/scijava/incubator/blob/a969114d2f047a47dffb3ca2d189a3263d56d98a/scijava-ops-image/src/main/java/org/scijava/ops/image/convert/clip/ClipRealTypes.java#L54-L66) Op that accepts a `min` and `max`, and make all params `I`. As a `Computer<T extends Type>` and/or `Function<T extends Comparable>`

**Comment by hinerm** [comment]
It turns out there **is** a suitable op: [`image.normalize`](https://github.com/scijava/incubator/tree/170852d2881d0a8eb5fa8245537583d4ceb062da/scijava-ops-image/src/main/java/org/scijava/ops/image/image/normalize).

It works for my use case. I do find it a little unwieldy. I'd like to see overriding signatures variations/optionals, e.g.:
-[ ] No input `sourceMin`/`sourceMax`, instead using `stats.min` and `stats.max` to get the display min/max values
-[ ] No `targetMin`/`targetMin`, instead having a signature that accepts an `O` and uses the `getMinValue`/`getMaxValue` of that type.
-[ ] Could also just omit the `O` and pull it from the `output` in the case of a computer

None of these are absolutely required, just sugar that I think would be nice for users
